src/net/Lib/Messaging/Providers/ServiceBusQueueMessagingProvider.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.ServiceBus;
+using Lib.Data;
+
+namespace Lib.Messaging.Providers
+{
+    public class ServiceBusQueueMessagingProvider : IMessagingProvider
+    {
+        public IQueue ImageQueueClient { get; set; }
+        public IQueue ClientSyncQueueClient { get; set; }
+
+        public ServiceBusQueueMessagingProvider()
+        {
+        }
+
+        public async Task InitializeAsync(TokenCredential credential, IDataProvider dataProvider)
+        {
+            await Task.Run(() =>
+            {
+                var serviceBusQueueClient = new ServiceBusClient(Config.ServiceBusNamespace, credential);

```

Comment:
How does this get closed/disposed?

Reply:
It doesn't currently.  Do we have a good example on what we expect people to do.  Just a note, I didn't think this would be Disposable since most of the other clients I've worked with are not.

Reply:
Unlike the HTTP-based services, Service Bus has to maintain an active connection to the service for communication over AMQP.   Disposing the client will trigger the active AMQP links, session, and connection to be cleaned up.   Otherwise, they'll remain active until the service boots them out;  during this time, they're still considered "alive" and count towards quotas and, may have billing impact in some hosting scenarios.

Looks like Service Bus did away with the explicit `CloseAsync` in favor of only supporting `IAsyncDisposable`.    Given the flow of your code, you're best bet would be to cache the client until you no longer need it and then manually call `DisposeAsync`.     You don't want to create/destroy it frequently while you're actually using it or you'll kill performance.   I don't see that Service Bus has gotten their documentation story fleshed out yet, but there's guidance in the [Event Hubs samples](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample02_EventHubsClients.md#lifetime) that discusses it, and we repeat it frequently in the API docs and other samples.